app/src/common/shared/org/mozilla/vrbrowser/telemetry/GleanMetricsService.java

```diff
@@ -0,0 +1,63 @@
+package org.mozilla.vrbrowser.telemetry;
+
+import android.content.Context;
+import android.util.Log;
+
+import androidx.annotation.UiThread;
+
+import org.mozilla.vrbrowser.browser.SettingsStore;
+import org.mozilla.vrbrowser.utils.DeviceType;
+import org.mozilla.vrbrowser.utils.SystemUtils;
+import org.mozilla.vrbrowser.BuildConfig;
+import org.mozilla.vrbrowser.GleanMetrics.Metrics;
+
+import mozilla.components.service.glean.Glean;
+import mozilla.components.service.glean.config.Configuration;
+
+
+public class GleanMetricsService {
+
+    private final static String APP_NAME = "FirefoxReality";
+    private static boolean initialized = false;
+    private final static String LOGTAG = SystemUtils.createLogtag(GleanMetricsService.class);
+    private static Context context = null;
+
+    // We should call this at the application initial stage.
+    public static void init(Context aContext) {
+        if (initialized)
+            return;
+
+        context = aContext;
+        initialized = true;
+
+        final boolean telemetryEnabled = SettingsStore.getInstance(aContext).isTelemetryEnabled();
+        Glean.INSTANCE.setUploadEnabled(telemetryEnabled);
+        Configuration config = new Configuration(Configuration.DEFAULT_TELEMETRY_ENDPOINT,
+                                                 BuildConfig.BUILD_TYPE);
+        Glean.INSTANCE.initialize(aContext, config);
+
+        if (!Glean.INSTANCE.isInitialized()) {
+            Log.d(LOGTAG, "Glean doesn't be initialized yet.");
+        }
+
+    }
+
+    // It would be called when users turn on/off the setting of telemetry.
+    // e.g., SettingsStore.getInstance(context).setTelemetryEnabled();
+    @UiThread
+    public static void start() {
+        Glean.INSTANCE.setUploadEnabled(true);
+        setStartupMetrics();
+    }
+
+    // It would be called when users turn on/off the setting of telemetry.
+    // e.g., SettingsStore.getInstance(context).setTelemetryEnabled();
+    @UiThread
+    public static void stop() {
+        Glean.INSTANCE.setUploadEnabled(false);
+    }
+
+    private static void setStartupMetrics() {
+        Metrics.INSTANCE.distributionName.set(APP_NAME + "_" + (DeviceType.isOculusBuild() ? "oculusvr" : BuildConfig.FLAVOR_platform));
+    }
```

Comment:
@Dexterp37 It is still not very clear about how to use Metric APIs from Java. Can you show me some examples when setting `distributionName`?

Reply:
> 
> 
> @Dexterp37 It is still not very clear about how to use Metric APIs from Java. Can you show me some sample code when setting `distributionName`?

The API looks slightly different in Java, but IDE autocompletion suggested me the following which works:

`Metrics.INSTANCE.getDistributionName().set(APP_NAME + "_" + (DeviceType.isOculusBuild() ? "oculusvr" : BuildConfig.FLAVOR_platform));`

I filed [this bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1588060) to make the API a tiny bit nicer in Java, and [this bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1588062) for adding Java examples to our documentation.

> I didn't choose to use a custom ping because it seems like a metrics ping can achieve what I want.

Yes, custom pings would not have helped you in this case.